app/volume.go

```diff
@@ -737,3 +772,75 @@ func (job *Job) findARandomReadyNode(v *longhornclient.Volume) (string, error) {
 	}
 	return readyNodeList[rand.Intn(len(readyNodeList))], nil
 }
+
+func filterSnapshots(snapshots []longhornclient.Snapshot, predicate func(snapshot longhornclient.Snapshot) bool) []longhornclient.Snapshot {
+	filtered := []longhornclient.Snapshot{}
+	for _, snapshot := range snapshots {
+		if predicate(snapshot) {
+			filtered = append(filtered, snapshot)
+		}
+	}
+	return filtered
+}
+
+// filterSnapshotsWithLabel return snapshots that have LabelKey and LabelValue
+func filterSnapshotsWithLabel(snapshots []longhornclient.Snapshot, labelKey, labelValue string) []longhornclient.Snapshot {
+	return filterSnapshots(snapshots, func(snapshot longhornclient.Snapshot) bool {
+		snapshotLabelValue, found := snapshot.Labels[labelKey]
+		if found && labelValue == snapshotLabelValue {
+			return true
+		}
+		return false
+	})
+}
+
+// filterSnapshotsNotInTargets returns snapshots that are not in the Targets
+func filterSnapshotsNotInTargets(snapshots []longhornclient.Snapshot, targets map[string]struct{}) []longhornclient.Snapshot {
+	return filterSnapshots(snapshots, func(snapshot longhornclient.Snapshot) bool {
+		if _, ok := targets[snapshot.Name]; !ok {
+			return true
+		}
+		return false
+	})
+}
+
+// getOldNames returns a list of names from the input sts excluding the latest retainCount names
+func getOldNames(nts []NameWithTimestamp, retainCount int) []string {
+	sort.Slice(nts, func(i, j int) bool {
+		if nts[i].Timestamp.Before(nts[j].Timestamp) {
```

Comment:
you can just do `return nts[i].Timestamp.Before(nts[j].Timestamp)`

Reply:
Thank you. Fixed